services/blue/src/api/Validation.cs

```diff
@@ -43,6 +43,24 @@ public static void Valid(SelectRoomType message)
             if (message.End < message.Start)
                 throw new ValidationException($"'{nameof(message.Start)}' must be before '{nameof(message.End)}'.");
         }
+
+        public static void Valid(GetRoomNumberRequest message)
+        {
+            if (message is null)
+                throw new NullReferenceException(nameof(ConfirmSelectedRoomType));
```

Comment:
NRE would be thrown automatically, so why bother?

```suggestion
                throw new ArgumentNullException(nameof(message));
```